ui/src/app/components/apps-panel/expanded-app.spec.tsx

```diff
@@ -0,0 +1,290 @@
+const onClickDeleteRuntime = jest.fn();
+
+const component = async (
+  appType: UIAppType,
+  initialUserAppInfo: UserAppEnvironment
+) =>
+  mount(
+    <ExpandedApp
+      {...{
+        appType,
+        initialUserAppInfo,
+        workspace,
+        onClickRuntimeConf,
+        onClickDeleteRuntime,
+      }}
+    />
+  );
+
+describe('ExpandedApp', () => {
+  const appsStub = new AppsApiStub();
+  const runtimeStub = new RuntimeApiStub();
+  runtimeStub.runtime.googleProject = googleProject;
+  beforeEach(() => {
+    serverConfigStore.set({ config: defaultServerConfig });
+    registerApiClient(AppsApi, appsStub);
+    registerApiClient(NotebooksApi, new NotebooksApiStub());
+    leoRegisterApiClient(LeoAppsApi, new LeoAppsApiStub());
+    leoRegisterApiClient(RuntimesApi, new RuntimesApiStub());
+    runtimeStore.set({
+      workspaceNamespace: workspace.namespace,
+      runtime: runtimeStub.runtime,
+      runtimeLoaded: true,
+    });
+  });
+
+  test.each([
+    ['should', RuntimeStatus.Running, true],
+    ['should', RuntimeStatus.Stopped, true],
+
+    ['should not', RuntimeStatus.Stopping, false],
+    ['should not', RuntimeStatus.Starting, false],
+    ['should not', RuntimeStatus.Error, false],
+    ['should not', RuntimeStatus.Unknown, false],
+    ['should not', undefined, false],
+    ['should not', null, false],
+  ])(
+    '%s allow deletion when the Jupyter app status is %s',
+    async (shouldOrNot, status, expectedCanDelete) => {
+      runtimeStub.runtime.status = status;
+
+      const wrapper = await component(UIAppType.JUPYTER, undefined);
+      expect(wrapper.exists()).toBeTruthy();
+
+      const deletion = wrapper.find({
+        'data-test-id': 'Jupyter-delete-button',
+      });
+      expect(deletion.exists()).toBeTruthy();
+      const { disabled } = deletion.props();
+
+      if (expectedCanDelete) {
+        expect(disabled).toBeFalsy();
+
+        const { onClick } = deletion.props();
+        await onClick();
+
+        expect(onClickDeleteRuntime).toHaveBeenCalled();
+      } else {
+        expect(disabled).toBeTruthy();
+      }
+    }
+  );
+
+  test.each([
+    ['should', 'Pause', RuntimeStatus.Running, true, false],
+    ['should', 'Resume', RuntimeStatus.Stopped, false, true],
+
+    ['should not', 'Pausing', RuntimeStatus.Stopping, false, false],
+    ['should not', 'Resuming', RuntimeStatus.Starting, false, false],
+
+    ['should not', 'Pause', RuntimeStatus.Deleted, false, false],
+    ['should not', 'Pause', RuntimeStatus.Deleting, false, false],
+    ['should not', 'Pause', RuntimeStatus.Unknown, false, false],
+    ['should not', 'Pause', RuntimeStatus.Error, false, false],
+    ['should not', 'Pause', RuntimeStatus.Creating, false, false],
+    ['should not', 'Pause', RuntimeStatus.Updating, false, false],
+
+    ['should not', 'Pause', undefined, false, false],
+    ['should not', 'Pause', null, false, false],
+  ])(
+    '%s allow clicking %s when the Jupyter app status is %s',
```

Comment:
I think you've gone too far with this one 😆 

I think this should be broken into a few separate tests to make it easier to understand; Something like `should allow pausing the app when the Jupyter status is X`.

Same with the below tests. I prefer to avoid `if-else` statements in tests unless it's absolutely needed.